### ISSUE TITLE
[collapsible-menu-view] Add additional styling, alignment options for collapsible-menu-view

### ISSUE THREAD

**Issue by kschuste** [cerner/terra-framework#1444]
# Feature Request

## Description
Am needing the ability to customize the collapsible menu view that are as follows:
1.  Allowing to specify left or right alignment, am needing left alignment
2. Allow to pass in the different variant styles from terra-button for the menu items, specify de-emphasis.
3. Allow to pass in the variant style for the dropdown button
4. Allow to give our own icon for the dropdown button, use a dropdown arrow instead of ...
5. Allow to specify if the dropdown button should be iconOnly, if don't want to use the 'More' text 

Proposal is to expose 5 additional new props to achieve this:

horizontalAlign: PropTypes.oneOf(['right', 'left']),
menuItemButtonVariant:  PropTypes.oneOf([Button.Opts.Variants....]),
menuItemDropdownButtonVariant:  PropTypes.oneOf([Button.Opts.Variants....]),
menuItemDropdownButtonIcon: PropTypes.element,
menuItemDropdownButtonIsIconOnly: PropTypes.bool,

Would then could pass the button variant stylings on down to the terra-button's that are used.
Would also need to expose the buttonVariant for the terra-button-group as well to style them the same given the variant.  Issue logged for that here:  https://github.com/cerner/terra-core/issues/3476

## Additional Context / Screenshots
<img width="574" alt="Screen Shot 2021-08-06 at 8 46 23 AM" src="https://user-images.githubusercontent.com/1711642/128551717-e398632a-5365-4f6b-be53-e057e5bef2ce.png">
